### PR TITLE
Add doctrine proxy support

### PR DESCRIPTION
**Pull request by TiMESPLiNTER**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -20,7 +20,8 @@
         "phpunit/phpunit": "~6.0",
         "psr/container": "~1.0",
         "pimple/pimple": "~3.0",
-        "symfony/yaml": "~2.7|~3.0"
+        "symfony/yaml": "~2.7|~3.0",
+        "doctrine/common": "^2.8.1"
```
Comment (dominikzogg) [inline composer.json]: @TiMESPLiNTER please use lowest supported version, not the newest once

File: src/Accessor/PropertyAccessor.php [hunk 2]
```diff
@@ -28,7 +29,11 @@ public function __construct(string $property)
      */
     public function getValue($object)
     {
-        $class = get_class($object);
+        if (true === class_exists('Doctrine\Common\Persistence\Proxy') && true === $object instanceof Proxy) {
```
Comment (dominikzogg) [inline src/Accessor/PropertyAccessor.php]: @TiMESPLiNTER please do not use true === if the return value of a method is a boolean

File: src/Normalizer/NormalizerObjectMappingRegistry.php [hunk 3]
```diff
@@ -42,6 +42,12 @@ private function addObjectMapping(NormalizationObjectMappingInterface $objectMap
      */
     public function getObjectMapping(string $class): NormalizationObjectMappingInterface
     {
+        $ref = new \ReflectionClass($class);
+
+        if (true === in_array('Doctrine\Common\Persistence\Proxy', $ref->getInterfaceNames(), true)) {
```
Comment (dominikzogg) [inline src/Normalizer/NormalizerObjectMappingRegistry.php]: @TiMESPLiNTER please do not use true === if the return value of a method is a boolean

File: src/Normalizer/Relation/EmbedManyFieldNormalizer.php [hunk 4]
```diff
@@ -51,6 +52,12 @@ public function normalizeField(
 
         $values = [];
         foreach ($childObjects as $i => $childObject) {
+            if (true === class_exists('Doctrine\Common\Persistence\Proxy')
```
Comment (dominikzogg) [inline src/Normalizer/Relation/EmbedManyFieldNormalizer.php]: @TiMESPLiNTER please do not use true === if the return value of a method is a boolean

File: src/Normalizer/Relation/EmbedOneFieldNormalizer.php [hunk 5]
```diff
@@ -49,6 +50,12 @@ public function normalizeField(
             return null;
         }
 
+        if (true === class_exists('Doctrine\Common\Persistence\Proxy')
```
Comment (dominikzogg) [inline src/Normalizer/Relation/EmbedOneFieldNormalizer.php]: @TiMESPLiNTER please do not use true === if the return value of a method is a boolean

**Review by dominikzogg — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (TiMESPLiNTER) [reply composer.json]: Done.

**Review by dominikzogg — APPROVED** [review]

**Pull request merged**